### ISSUE TITLE
Resolver docs contradict ts docs, or does it?

### ISSUE THREAD

**Issue by antoniormrzz** [MichalLytek/type-graphql#756]
**Describe the issue**
The resolver docs use an optional parameter before a required parameter. ts docs say that "Any optional parameters must follow required parameters."
the part in question is this (arguments section):
```
@Resolver()
class RecipeResolver {
  // ...
  @Query(returns => [Recipe])
  async recipes(
    @Arg("title", { nullable: true }) title?: string,
    @Arg("servings", { defaultValue: 2 }) servings: number,
  ): Promise<Recipe[]> {
    // ...
  }
}
```
servings does have defaultValue, so it is optional in graphql, however that is not the case in ts example.
Maybe I'm getting confused.

**Are you able to make a PR that fix this?**
I could, would love to.

**Comment by antoniormrzz** [comment]
if you see fit, I would love to have the honor of contribution 😄

**Comment by MichalLytek** [comment]
You're right, it should be in reverse order. Or you can just do `title: string | undefined` 😉 
```ts
@Resolver()
class RecipeResolver {
  // ...
  @Query(returns => [Recipe])
  async recipes(
    @Arg("servings", { defaultValue: 2 }) servings: number,
    @Arg("title", { nullable: true }) title?: string,
  ): Promise<Recipe[]> {
    // ...
  }
}
```

> I would love to have the honor of contribution

Please go ahead 😄

**Comment by MichalLytek** [comment]
Closed by #757
